internal/telemetry/reporter.go

```diff
@@ -0,0 +1,226 @@
+// Package telemetry implements a client for reporting telemetry data used to
+// prioritize development of SpiceDB.
+//
+// For more information, see:
+// https://github.com/authzed/spicedb/blob/main/TELEMETRY.md
+package telemetry
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"strconv"
+	"time"
+
+	"github.com/cenkalti/backoff/v4"
+	"github.com/gogo/protobuf/proto"
+	"github.com/golang/snappy"
+	dto "github.com/prometheus/client_model/go"
+	"github.com/prometheus/common/model"
+	"github.com/prometheus/prometheus/prompb"
```

Comment:
We can drop importing all of prometheus if we use this:

`go.buf.build/protocolbuffers/go/prometheus/prometheus`